haas/cli.py

```diff
@@ -179,13 +195,16 @@ def setup_http_client():
 
     This may be extended with other backends in the future.
     """
-    global http_client
```

Comment:
@SahilTikale, you still haven't addressed this (though the comment has been hidden because you changed code nearby).

There are a number of other comments that I made in the past that you've not addressed -- please pick through them and do so.